contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,836 @@
+                WETHAmountSupplied, // WETHAmountSupplied
+                OETHbAmountSupplied, // OETHbAmountSupplied
+                tokenId, // tokenId
+                netValue
+            );
+        }
+
+        // burn remaining OETHb
+        _burnOETHbOnTheContract();
+        positionManager.approve(address(clGauge), tokenId);
+        clGauge.deposit(tokenId);
+    }
+
+    /**
+     * @dev Check that the liquidity in the pool is withing the expected WETH to OETHb ratio
+     */
+    function _checkLiquidityWithinExpectedShare() internal {
+        (uint256 WETHPositionBalance, uint256 OETHbPositionBalance) = getPositionPrincipal();
+        // TODO check we are withing the expected tick range
+        require(WETHPositionBalance + OETHbPositionBalance > 0, "Can not withdraw full position");
+
+        uint256 currentWethShareBp = 0;
+        if (WETHPositionBalance != 0) {
+            currentWethShareBp = WETHPositionBalance * 1e4 / (WETHPositionBalance + OETHbPositionBalance);
+        }
+
+        if (currentWethShareBp < poolWethShareVarianceAllowed || // uint256 musn't go below 0  
+            poolWethShare < currentWethShareBp - poolWethShareVarianceAllowed ||
+            poolWethShare > currentWethShareBp + poolWethShareVarianceAllowed) {
+
+            revert PoolRebalanceOutOfBounds(
+                currentWethShareBp,
+                poolWethShare,
+                WETHPositionBalance,
+                OETHbPositionBalance
+            );
+        } else {
+            emit PoolRebalanced(
+                currentWethShareBp,
+                poolWethShare,
+                WETHPositionBalance,
+                OETHbPositionBalance
+            );
+        }
+    }
+
+    /**
+     * Burns any lingering OETHb tokens still remaining on the contract
+     */
+    function _burnOETHbOnTheContract() internal {
+        uint256 OETHbBalance = IERC20(OETHb).balanceOf(address(this));
+        IVault(vaultAddress).burnForStrategy(OETHbBalance);
+    }
+
+    function updateNetValue() internal {
+        if (tokenId == 0) {
+            netValue = 0;
+        } else {
+            (uint128 liquidity,,) = _getPositionInfo();
+
+            /**
+             * Our net value represent the smallest amount of tokens we are able to extract from the position
+             * given our liquidity.
+             *
+             * The least amount of tokens extraditable from the position is where the active trading price is
+             * at the ticker 0 meaning the pool is offering 1:1 trades between WETH & OETHb. At that moment the pool 
+             * consists completely of OETHb and no WETH.
+             * 
+             * The more swaps from WETH -> OETHb happen on the pool the more the price starts to move towards the -1
+             * ticker making OETHb (priced in WETH) more expensive.
+             */
+            (uint256 wethAmount, uint256 OETHbAmount) = helper.getAmountsForLiquidity(
```

Comment:
🔴 This method of calculating is great because it works even if we using tick bands that didn't touch zero. Very cool. However it does make an assumption about which side of the ticks is the one closest to 1:1, which could give the wrong answer when we use this same AMO contract against another coin.

Reply:
Oh awesome thinking. Way to prevent very hard to detect errors in the future. Fix for this here: https://github.com/OriginProtocol/origin-dollar/pull/2184/commits/360240755dadaa27fc1353914e8b0855098e31c3